www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,442 @@
+  const { mineralSiteStore, dedupMineralSiteStore, userStore, commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore } = useStores();
+  const [form] = Form.useForm();
+  const [recordId, setRecordId] = useState("");
+  const [selectedSourceType, setSelectedSourceType] = useState<string | null>(null);
+  const commodityOptions = useMemo(() => {
+    return Array.from(commodityStore.records.values()).map((comm) => ({
+      value: comm?.id,
+      label: comm?.name,
+    }));
+  }, [commodityStore.records]);
+  const currentUser = userStore.getCurrentUser()?.url;
+  const countryOptions = useMemo(() => {
+    return Array.from(countryStore.records.values()).map((country) => ({
+      value: country?.id,
+      label: country?.name,
+    }));
+  }, [countryStore.records]);
+
+  const stateOptions = useMemo(() => {
+    return Array.from(stateOrProvinceStore.records.values()).map((state) => ({
+      value: state?.id,
+      label: state?.name,
+    }));
+  }, [stateOrProvinceStore.records]);
+
+  const depositTypeOptions = useMemo(() => {
+    return Array.from(depositTypeStore.records.values()).map((type) => ({
+      value: type?.id ?? "",
+      label: type?.name ?? "Unnamed",
+    }));
+  }, [depositTypeStore.records]);
+
+  const unitOptions = useMemo(() => {
+    return Array.from(unitStore.records.values()).map((unit) => ({
+      value: unit?.id ?? "",
+      label: unit?.name ?? "Unnamed",
+    }));
+  }, [unitStore.records]);
+  useEffect(() => {
+    const initialize = async () => {
+      try {
+
+        if (visible) {
+          const generatedId = `record-${uuidv4()}`;
+          setRecordId(generatedId)
+          form.setFieldsValue({ recordId: generatedId });
+        }
+        console.log("Generated ID:", recordId);
+      } catch (error) {
+        console.error("Error initializing stores:", error);
+      }
+    };
+
+    initialize();
+  }, [form, visible]);
+
+  const handleSourceTypeChange = (e: RadioChangeEvent) => {
+    const value = e.target.value;
+    setSelectedSourceType(value);
+
+    const refDocUrl = form.getFieldValue("refDoc");
+    if (value === "unpublished") {
+      form.setFieldsValue({ sourceId: `unpublished::${currentUser || "unknown"}` });
+    } else if (refDocUrl) {
+      form.setFieldsValue({ sourceId: `${value}::${refDocUrl}` });
+    } else {
+      form.setFieldsValue({ sourceId: null });
+    }
+  };
+
+
+
+  const handleSave = async (values: FormValues) => {
+    try {
+      let location = undefined;
+      if (values.latitude !== undefined && values.longitude !== undefined) {
+        location = `POINT (${values.longitude} ${values.latitude})`;
+      }
+
+      const countries = values.country
+        ? [
+          new CandidateEntity({
+            observedName: countryStore.getByURI(values.country)?.name,
+            source: currentUser ? currentUser : "unknown",
```

Comment:
currentUser is not null!

Reply:
webpack compiled with 1 warning
ERROR in src/pages/editor/NewMineralSiteModal.tsx:120:13
TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.
    118 |           new CandidateEntity({
    119 |             observedName: countryStore.getByURI(values.country)?.name,
  > 120 |             source: currentUser,
        |             ^^^^^^
    121 |             normalizedURI: values.country,
    122 |             confidence: 1.0,
    123 |           }),